piper/mouseperspective.py

```diff
@@ -83,20 +82,18 @@ def device(self):
 
     def set_device(self, device):
         self._device = device
-        capabilities = device.capabilities
         device.connect("resync", lambda _: self._show_notification_error())
 
         self.stack.foreach(Gtk.Widget.destroy)
-        if RatbagdDevice.CAP_RESOLUTION in capabilities:
+        active_profile = device.active_profile
+        if len(active_profile.resolutions) > 0:
             self.stack.add_titled(ResolutionsPage(device), "resolutions", _("Resolutions"))
-        if RatbagdDevice.CAP_BUTTON in capabilities:
+        if len(active_profile.buttons) > 0:
             self.stack.add_titled(ButtonsPage(device), "buttons", _("Buttons"))
-        if RatbagdDevice.CAP_LED in capabilities:
+        if len(active_profile.leds) > 0:
```

Comment:
fwiw, the pythonic way here is a simple `if active_profile.leds:` because an empty list is false. The main reason this is superior is efficiency, `len()` has to count the elements only for the result to be thrown away whereas the other option immediately returns when one element is found.

Reply:
Thanks, dabbling too much with other languages atm :)